exercises/gagik_vardanyan/ex02_oo_basics/src/main.cpp

```diff
@@ -14,22 +14,17 @@
 ///the main program
 int main(int argc, char ** argv)
 {
-    std::vector<Shape> shapes;
+    std::vector<Shape*> shapes;
 
     Rectangle * rect = new Rectangle(10,50);
```

Comment:
Good. However this will leak resources, because you don't ever `delete` the contents of the `std::vector`. You could have solved this in a couple of different ways. 1. You could have introduced a clean-up that enumerated over the vector, and deleted each element. 2. You could have just declared the `Shape` objects on the stack and then just taken their address when filling the vector.

```cpp
Triangle tri(10, 5)
shapes.push_back(&tri);
```